lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -39,7 +39,12 @@ class DescribeClass < Cop
           (sym {:request :feature :routing :view}))
         PATTERN
 
+        def_node_matcher :shared_examples?, <<-PATTERN
+          (block (send (const nil :RSpec) :shared_examples ...) ...)
```

Comment:
We should probably support the following cases

```ruby
RSpec.shared_examples do
end

shared_examples do
end

RSpec.shared_context do
end

shared_context do
end
```

So I think the node pattern like this would look like so

```
(block (send {(const nil :RSpec) nil} {:shared_examples :shared_context : shared_examples_for} ...) ...)
```

I have `RuboCop::RSpec::Language` which defines a lot of the RSpec public interface: https://github.com/backus/rubocop-rspec/blob/master/lib/rubocop/rspec/language.rb

with that you can do

```ruby
<<-PATTERN
(block (send {(const nil :RSpec) nil} {#{SharedGroups::ALL.to_node_pattern} ...) ...)
PATTERN
```